src/apps/statements/repo/modelsRepo/updateFullActivities/mongo.ts

```diff
@@ -1,47 +1,107 @@
+  const extensions = replaceDotsInExtensions(/\./g, '&46;')(fullActivity.extensions);
+
+  return {
+    ...getPatchUpdate(fullActivity.name, ['name']),
+    ...getPatchUpdate(fullActivity.description, ['description']),
+    ...getPatchUpdate(extensions, ['extensions']),
+    ...(
+      fullActivity.moreInfo !== undefined
+        ? { moreInfo: fullActivity.moreInfo }
+        : {}
+    ),
+    ...(
+      fullActivity.type !== undefined
+        ? { type: fullActivity.type }
+        : {}
+    ),
+  };
+};
+
+const createMongoQuery = (fullActivity: FullActivityDatabase) => {
+  const activityId = fullActivity.activityId;
+  const lrsId = new ObjectID(fullActivity.lrsId);
+  const organisationId = new ObjectID(fullActivity.organisationId);
+
+  return matchesFullActivity({ activityId, lrsId, organisationId });
+};
+
+const creatBatchQuery = (batch: UnorderedBulkOperation) => (fullActivity: FullActivityDatabase) => {
+  const mongoQuery = createMongoQuery(fullActivity);
+  // tslint:disable:no-inferred-empty-object-type
+  const mongoSet = createMongoSet(fullActivity);
+  const mongoAddToSet = createMongoAddToSet(fullActivity);
+  // tslint:enable:no-inferred-empty-object-type
+
+  batch.find(mongoQuery).upsert().updateOne({
+    ...(!isEmpty(mongoSet) ? { $set: mongoSet } : {}),
+    ...(!isEmpty(mongoAddToSet) ? { $addToSet: mongoAddToSet } : {}),
   });
 };
 
-export default (config: FacadeConfig): Signature => {
-  return async ({ client, updates }) => {
+export default (config: FacadeConfig): Signature =>
+  async ({ fullActivities }) => {
+    if (fullActivities.length < 1) {
+      return;
+    }
+
     const collection = (await config.db()).collection(FULL_ACTIVITIES_COLLECTION_NAME);
-    const lrsId = new ObjectID(client.lrs_id);
-    const organisationId = new ObjectID(client.organisation);
     const batch = collection.initializeUnorderedBulkOp();
 
-    updates.forEach((update) => {
-      const activityId = update.activityId;
-      const extensions = replaceDotsInExtensions(/\./g, '&46;')(update.extensions);
-      const mongoQuery = matchesFullActivity({ activityId, lrsId, organisationId });
-      const mongoSet = {
-        ...getPatchUpdate(update.name, ['name']),
-        ...getPatchUpdate(update.description, ['description']),
-        ...getPatchUpdate(extensions, ['extensions']),
-        ...(update.moreInfo !== undefined ? { moreInfo: update.moreInfo } : {}),
-        ...(update.type !== undefined ? { type: update.type } : {}),
-      };
-      if (Object.keys(mongoSet).length === 0) {
-        /* istanbul ignore next */
-        return;
-      }
-      const mongoUpdate = { $set: mongoSet };
-
-      batch.find(mongoQuery).upsert().updateOne(mongoUpdate);
-    });
-
-    if (updates.length > 0) {
-      await batch.execute();
-    }
+    fullActivities.forEach(creatBatchQuery(batch));
+    await batch.execute();
+    // tslint:disable-next-line:max-file-line-count
```

Comment:
@PrinceWaune we don't use such ignore lines in other modules within the xapi-service - is this required?

Reply:
`tslint:disable-next-line:max-file-line-count` we use in all files that have more than 100 strings of the code.
In another case, we need to refactor all these files. So I think we don't need to do these changes here.

Reply:
Interesting, cheers @PrinceWaune !